qutebrowser/keyinput/modeparsers.py

```diff
@@ -27,7 +27,7 @@
 
 from qutebrowser.utils import message
 from qutebrowser.config import config
-from qutebrowser.keyinput import keyparser
+from qutebrowser.keyinput import keyparser, modeman
```

Comment:
You introduced a circular import here, which is why the tests are failing:

```
[...]
  File "/home/travis/build/The-Compiler/qutebrowser/tests/helpers/stubs.py", line 31, in <module>
    from qutebrowser.browser import webview
  File "/home/travis/build/The-Compiler/qutebrowser/qutebrowser/browser/webview.py", line 33, in <module>
    from qutebrowser.keyinput import modeman
  File "/home/travis/build/The-Compiler/qutebrowser/qutebrowser/keyinput/modeman.py", line 27, in <module>
    from qutebrowser.keyinput import modeparsers, keyparser
  File "/home/travis/build/The-Compiler/qutebrowser/qutebrowser/keyinput/modeparsers.py", line 30, in <module>
    from qutebrowser.keyinput import keyparser, modeman
ImportError: cannot import name 'modeman'
```

I'd suggest adding a `pyqtSignal` to `MarkKeyParser` instead of calling `modeman.leave` directly, and then connecting that signal in `ModeManager.register`.